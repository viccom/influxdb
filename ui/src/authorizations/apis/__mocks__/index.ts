import {authorization} from './data'

// Types
import {Authorization} from 'src/api'

export const getAuthorizations = async (): Promise<Authorization[]> => {
  return Promise.resolve([authorization, {...authorization, id: '1'}])
}
