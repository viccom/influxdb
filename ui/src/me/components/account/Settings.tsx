// Libraries
import React, {PureComponent, ChangeEvent} from 'react'
import {connect} from 'react-redux'

// Types
import {MeState} from 'src/types/v2'
import {
  Form,
  Button,
  Input,
  ComponentSize,
  ComponentStatus,
  Panel,
  Grid,
  Columns,
} from 'src/clockface'

interface StateProps {
  me: MeState
}

interface State {
  me: MeState
}

export class Settings extends PureComponent<StateProps, State> {
  constructor(props) {
    super(props)
    this.state = {
      me: this.props.me,
    }
  }

  public render() {
    const {me} = this.state

    return (
      <Grid>
        <Grid.Row>
          <Grid.Column widthXS={Columns.Six}>
            <Panel>
              <Panel.Header title="About Me">
                <Button text="Edit About Me" />
              </Panel.Header>
              <Panel.Body>
                <Form>
                  <Form.Element label="Username">
                    <Input
                      value={me.name}
                      dataTest="nameInput"
                      titleText="Username"
                      size={ComponentSize.Small}
                      status={ComponentStatus.Disabled}
                      onChange={this.handleChangeInput}
                    />
                  </Form.Element>
                </Form>
              </Panel.Body>
            </Panel>
          </Grid.Column>
        </Grid.Row>
      </Grid>
    )
  }

  private handleChangeInput = (_: ChangeEvent<HTMLInputElement>): void => {
    //  console.log('changing: ', e)
  }
}

const mstp = ({me}) => ({
  me,
})

export default connect<StateProps>(mstp)(Settings)
