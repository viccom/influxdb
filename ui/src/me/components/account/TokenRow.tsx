// Libraries
import React, {PureComponent} from 'react'

// Components
import {
  IndexList,
  ComponentSpacer,
  Alignment,
  Button,
  ComponentSize,
  ComponentColor,
} from 'src/clockface'

// Types
import {Authorization} from 'src/api'

interface Props {
  auth: Authorization
  onClickDescription: (authID: string) => void
}

export default class TokenRow extends PureComponent<Props> {
  public render() {
    const {description, status, id} = this.props.auth

    return (
      <IndexList.Row>
        <IndexList.Cell>
          <a
            href="#"
            onClick={this.handleClickDescription}
            data-test={`token-description-${id}`}
          >
            {description}
          </a>
        </IndexList.Cell>
        <IndexList.Cell>{status}</IndexList.Cell>
        <IndexList.Cell alignment={Alignment.Right} revealOnHover={true}>
          <ComponentSpacer align={Alignment.Right}>
            <Button
              size={ComponentSize.ExtraSmall}
              color={ComponentColor.Danger}
              text="Delete"
            />
          </ComponentSpacer>
        </IndexList.Cell>
      </IndexList.Row>
    )
  }

  private handleClickDescription = () => {
    const {onClickDescription, auth} = this.props
    onClickDescription(auth.id)
  }
}
