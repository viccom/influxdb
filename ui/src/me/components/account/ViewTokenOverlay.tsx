// Libraries
import React, {PureComponent} from 'react'

// Components
import {OverlayContainer, OverlayBody, OverlayHeading} from 'src/clockface'
import PermissionsWidget, {
  PermissionsWidgetMode,
  PermissionsWidgetSelection,
} from 'src/shared/components/permissionsWidget/PermissionsWidget'
import CopyText from 'src/shared/components/CopyText'

// Types
import {Authorization, Permission} from 'src/api'

// Actions
import {NotificationAction} from 'src/types'

const {Write, Read} = Permission.ActionEnum

interface Props {
  onNotify: NotificationAction
  auth: Authorization
  onDismissOverlay: () => void
}

const actions = [Read, Write]

export default class ViewTokenOverlay extends PureComponent<Props> {
  public render() {
    const {description, permissions} = this.props.auth
    const {onNotify} = this.props

    return (
      <OverlayContainer>
        <OverlayHeading title={description} onDismiss={this.handleDismiss} />
        <OverlayBody>
          <CopyText copyText={this.props.auth.token} notify={onNotify} />
          <PermissionsWidget
            mode={PermissionsWidgetMode.Read}
            heightPixels={500}
          >
            {permissions.map((p, i) => {
              return (
                <PermissionsWidget.Section
                  key={i}
                  id={this.id(p)}
                  title={this.title(p)}
                  mode={PermissionsWidgetMode.Read}
                >
                  {actions.map((a, i) => (
                    <PermissionsWidget.Item
                      key={i}
                      id={this.itemID(p, a)}
                      label={a}
                      selected={this.selected(p, a)}
                    />
                  ))}
                </PermissionsWidget.Section>
              )
            })}
          </PermissionsWidget>
        </OverlayBody>
      </OverlayContainer>
    )
  }

  private selected = (
    permission: Permission,
    action: Permission.ActionEnum
  ): PermissionsWidgetSelection => {
    if (permission.action === action) {
      return PermissionsWidgetSelection.Selected
    }

    return PermissionsWidgetSelection.Unselected
  }

  private itemID = (
    permission: Permission,
    action: Permission.ActionEnum
  ): string => {
    return `${permission.resource.type}-${action}-${permission.resource.id ||
      '*'}-${permission.resource.orgID || '*'}`
  }

  private id = (permission: Permission): string => {
    return permission.resource.type
  }

  private title = (permission: Permission): string => {
    const org = permission.resource.org || '*'
    const name = permission.resource.name || '*'
    const type = permission.resource.type

    if (permission.resource.name || permission.resource.org) {
      return `${org}:${type}:${name}`
    }

    return `${permission.resource.type}:*`
  }

  private handleDismiss = () => {
    this.props.onDismissOverlay()
  }
}
