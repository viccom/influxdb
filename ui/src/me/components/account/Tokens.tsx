// Libraries
import React, {PureComponent, ChangeEvent} from 'react'
import {connect} from 'react-redux'

// Components
import {IconFont, Input, Spinner} from 'src/clockface'
import ResourceFetcher from 'src/shared/components/resource_fetcher'
import TokenList from 'src/me/components/account/TokensList'
import FilterList from 'src/shared/components/Filter'
import TabbedPageHeader from 'src/shared/components/tabbed_page/TabbedPageHeader'

// APIs
import {getAuthorizations} from 'src/authorizations/apis'

// Actions
import {notify} from 'src/shared/actions/notifications'
import {NotificationAction} from 'src/types'

// Types
import {Authorization} from 'src/api'

interface State {
  searchTerm: string
}

enum AuthSearchKeys {
  Description = 'description',
  Status = 'status',
}

interface Props {
  onNotify: NotificationAction
}

export class Tokens extends PureComponent<Props, State> {
  constructor(props) {
    super(props)
    this.state = {
      searchTerm: '',
    }
  }

  public render() {
    const {onNotify} = this.props
    const {searchTerm} = this.state

    return (
      <>
        <TabbedPageHeader>
          <Input
            icon={IconFont.Search}
            value={searchTerm}
            placeholder="Filter Tokens..."
            onChange={this.handleChangeSearchTerm}
            widthPixels={256}
          />
        </TabbedPageHeader>
        <ResourceFetcher<Authorization[]> fetcher={getAuthorizations}>
          {(fetchedAuths, loading) => (
            <Spinner loading={loading}>
              <FilterList<Authorization>
                list={fetchedAuths}
                searchTerm={searchTerm}
                searchKeys={this.searchKeys}
              >
                {filteredAuths => (
                  <TokenList
                    auths={filteredAuths}
                    onNotify={onNotify}
                    searchTerm={searchTerm}
                  />
                )}
              </FilterList>
            </Spinner>
          )}
        </ResourceFetcher>
      </>
    )
  }

  private handleChangeSearchTerm = (e: ChangeEvent<HTMLInputElement>): void => {
    this.setState({searchTerm: e.target.value})
  }

  private get searchKeys(): AuthSearchKeys[] {
    return [AuthSearchKeys.Status, AuthSearchKeys.Description]
  }
}

const mdtp = {
  onNotify: notify,
}

export default connect<Props>(
  null,
  mdtp
)(Tokens)
