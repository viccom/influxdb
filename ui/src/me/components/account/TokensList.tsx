// Libraries
import React, {PureComponent} from 'react'

// Components
import {
  IndexList,
  EmptyState,
  ComponentSize,
  OverlayTechnology,
} from 'src/clockface'
import TokenRow from 'src/me/components/account/TokenRow'

// Types
import {Authorization} from 'src/api'
import ViewTokenOverlay from './ViewTokenOverlay'

// Actions
import {NotificationAction} from 'src/types'

interface Props {
  auths: Authorization[]
  onNotify: NotificationAction
  searchTerm: string
}

interface State {
  isTokenOverlayVisible: boolean
  authInView: Authorization
}

export default class TokenList extends PureComponent<Props, State> {
  constructor(props) {
    super(props)
    this.state = {
      isTokenOverlayVisible: false,
      authInView: null,
    }
  }

  public render() {
    const {auths, onNotify} = this.props
    const {isTokenOverlayVisible, authInView} = this.state

    return (
      <>
        <IndexList>
          <IndexList.Header>
            <IndexList.HeaderCell columnName="Description" />
            <IndexList.HeaderCell columnName="Status" />
          </IndexList.Header>
          <IndexList.Body emptyState={this.emptyState} columnCount={2}>
            {auths.map(a => {
              return (
                <TokenRow
                  key={a.id}
                  auth={a}
                  onClickDescription={this.handleClickDescription}
                />
              )
            })}
          </IndexList.Body>
        </IndexList>
        <OverlayTechnology visible={isTokenOverlayVisible}>
          <ViewTokenOverlay
            onNotify={onNotify}
            auth={authInView}
            onDismissOverlay={this.handleDismissOverlay}
          />
        </OverlayTechnology>
      </>
    )
  }

  private handleDismissOverlay = () => {
    this.setState({isTokenOverlayVisible: false})
  }

  private handleClickDescription = (authID: string): void => {
    const authInView = this.props.auths.find(a => a.id === authID)
    this.setState({isTokenOverlayVisible: true, authInView})
  }

  private get emptyState(): JSX.Element {
    const {searchTerm} = this.props
    let emptyStateText =
      'There are not any Tokens associated with this account. Contact your administrator'

    if (searchTerm) {
      emptyStateText = 'No Tokens match your search term'
    }

    return (
      <EmptyState size={ComponentSize.Large}>
        <EmptyState.Text text={emptyStateText} />
      </EmptyState>
    )
  }
}
