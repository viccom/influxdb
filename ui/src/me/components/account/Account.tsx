// Libraries
import React, {PureComponent} from 'react'
import {WithRouterProps} from 'react-router'

// Components
import {Page} from 'src/pageLayout'
import TabbedPage from 'src/shared/components/tabbed_page/TabbedPage'
import TabbedPageSection from 'src/shared/components/tabbed_page/TabbedPageSection'
import Settings from 'src/me/components/account/Settings'
import Tokens from 'src/me/components/account/Tokens'

export enum Tabs {
  Settings = 'settings',
  Tokens = 'tokens',
}

export default class Account extends PureComponent<WithRouterProps> {
  public render() {
    const {params} = this.props
    return (
      <Page titleTag="Account">
        <Page.Header fullWidth={false}>
          <Page.Header.Left>
            <Page.Title title="Account" />
          </Page.Header.Left>
          <Page.Header.Right />
        </Page.Header>
        <Page.Contents fullWidth={false} scrollable={true}>
          <TabbedPage
            name="Account"
            activeTabUrl={params.tab}
            parentUrl="/account"
          >
            <TabbedPageSection
              title="My Settings"
              id={Tabs.Settings}
              url={Tabs.Settings}
            >
              <Settings />
            </TabbedPageSection>
            <TabbedPageSection
              title="My Tokens"
              id={Tabs.Tokens}
              url={Tabs.Tokens}
            >
              <Tokens />
            </TabbedPageSection>
          </TabbedPage>
        </Page.Contents>
      </Page>
    )
  }
}
