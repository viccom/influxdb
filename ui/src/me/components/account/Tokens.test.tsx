// Libraries
import React from 'react'
import {mount} from 'enzyme'

// Components
import {Tokens} from 'src/me/components/account/Tokens'
import TokenRow from 'src/me/components/account/TokenRow'
import ViewTokenModal from 'src/me/components/account/ViewTokenOverlay'

// Fixtures
import {authorization} from 'src/authorizations/apis/__mocks__/data'

jest.mock('src/authorizations/apis')

const setup = (override?) => {
  const props = {
    authorizationsLink: 'api/v2/authorizations',
    ...override,
  }

  const tokensWrapper = mount(<Tokens {...props} />)

  return {tokensWrapper}
}

describe('Account', () => {
  let wrapper
  beforeEach(done => {
    const {tokensWrapper} = setup()
    wrapper = tokensWrapper
    process.nextTick(() => {
      wrapper.update()
      done()
    })
  })

  describe('rendering', () => {
    it('renders!', () => {
      expect(wrapper.exists()).toBe(true)
      expect(wrapper).toMatchSnapshot()
    })

    it('displays the list of tokens', () => {
      const rows = wrapper.find(TokenRow)
      expect(rows.length).toBe(2)
    })
  })

  describe('user interaction', () => {
    describe('clicking the token description', () => {
      it('opens the ViewTokenModal', () => {
        const description = wrapper.find({
          'data-test': `token-description-${authorization.id}`,
        })
        description.simulate('click')
        wrapper.update()

        const modal = wrapper.find(ViewTokenModal)

        expect(modal.exists()).toBe(true)
      })
    })
  })
})
