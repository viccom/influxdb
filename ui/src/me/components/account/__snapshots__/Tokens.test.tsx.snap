// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account rendering renders! 1`] = `
<Tokens
  authorizationsLink="api/v2/authorizations"
>
  <TabbedPageHeader>
    <div
      className="tabbed-page-section--header"
    >
      <Input
        autoFocus={false}
        autocomplete="off"
        disabledTitleText="This input is disabled"
        icon="search"
        name=""
        onChange={[Function]}
        placeholder="Filter Tokens..."
        size="sm"
        spellCheck={false}
        status="default"
        titleText=""
        value=""
        widthPixels={256}
      >
        <div
          className="input input-sm input--has-icon"
          style={
            Object {
              "width": "256px",
            }
          }
        >
          <input
            autoComplete="off"
            autoFocus={false}
            className="input-field"
            disabled={false}
            name=""
            onChange={[Function]}
            placeholder="Filter Tokens..."
            spellCheck={false}
            title=""
            value=""
          />
          <span
            className="input-icon icon search"
          />
          <div
            className="input-shadow"
          />
        </div>
      </Input>
    </div>
  </TabbedPageHeader>
  <ResourceFetcher
    fetcher={[Function]}
  >
    <Spinner
      loading="Done"
    >
      <FilterList
        list={
          Array [
            Object {
              "description": "im a token",
              "id": "030444b11fb10000",
              "links": Object {
                "self": "/api/v2/authorizations/030444b11fb10000",
                "user": "/api/v2/users/030444b10a710000",
              },
              "orgID": "030444b10a713000",
              "permissions": Array [
                Object {
                  "action": "write",
                  "resource": Object {
                    "type": "orgs",
                  },
                },
                Object {
                  "action": "write",
                  "resource": Object {
                    "type": "buckets",
                  },
                },
              ],
              "status": "active",
              "token": "ohEmfY80A9UsW_cicNXgOMIPIsUvU6K9YcpTfCPQE3NV8Y6nTsCwVghczATBPyQh96CoZkOW5DIKldya6Y84KA==",
              "user": "watts",
              "userID": "030444b10a710000",
            },
            Object {
              "description": "im a token",
              "id": "1",
              "links": Object {
                "self": "/api/v2/authorizations/030444b11fb10000",
                "user": "/api/v2/users/030444b10a710000",
              },
              "orgID": "030444b10a713000",
              "permissions": Array [
                Object {
                  "action": "write",
                  "resource": Object {
                    "type": "orgs",
                  },
                },
                Object {
                  "action": "write",
                  "resource": Object {
                    "type": "buckets",
                  },
                },
              ],
              "status": "active",
              "token": "ohEmfY80A9UsW_cicNXgOMIPIsUvU6K9YcpTfCPQE3NV8Y6nTsCwVghczATBPyQh96CoZkOW5DIKldya6Y84KA==",
              "user": "watts",
              "userID": "030444b10a710000",
            },
          ]
        }
        searchKeys={
          Array [
            "status",
            "description",
          ]
        }
        searchTerm=""
      >
        <TokenList
          auths={
            Array [
              Object {
                "description": "im a token",
                "id": "030444b11fb10000",
                "links": Object {
                  "self": "/api/v2/authorizations/030444b11fb10000",
                  "user": "/api/v2/users/030444b10a710000",
                },
                "orgID": "030444b10a713000",
                "permissions": Array [
                  Object {
                    "action": "write",
                    "resource": Object {
                      "type": "orgs",
                    },
                  },
                  Object {
                    "action": "write",
                    "resource": Object {
                      "type": "buckets",
                    },
                  },
                ],
                "status": "active",
                "token": "ohEmfY80A9UsW_cicNXgOMIPIsUvU6K9YcpTfCPQE3NV8Y6nTsCwVghczATBPyQh96CoZkOW5DIKldya6Y84KA==",
                "user": "watts",
                "userID": "030444b10a710000",
              },
              Object {
                "description": "im a token",
                "id": "1",
                "links": Object {
                  "self": "/api/v2/authorizations/030444b11fb10000",
                  "user": "/api/v2/users/030444b10a710000",
                },
                "orgID": "030444b10a713000",
                "permissions": Array [
                  Object {
                    "action": "write",
                    "resource": Object {
                      "type": "orgs",
                    },
                  },
                  Object {
                    "action": "write",
                    "resource": Object {
                      "type": "buckets",
                    },
                  },
                ],
                "status": "active",
                "token": "ohEmfY80A9UsW_cicNXgOMIPIsUvU6K9YcpTfCPQE3NV8Y6nTsCwVghczATBPyQh96CoZkOW5DIKldya6Y84KA==",
                "user": "watts",
                "userID": "030444b10a710000",
              },
            ]
          }
          searchTerm=""
        >
          <IndexList>
            <table
              className="index-list"
            >
              <IndexListHeader>
                <thead
                  className="index-list--header"
                >
                  <tr>
                    <IndexListHeaderCell
                      alignment="left"
                      columnName="Description"
                    >
                      <th
                        className="index-list--header-cell index-list--align-left"
                        onClick={[Function]}
                        style={
                          Object {
                            "width": undefined,
                          }
                        }
                      >
                        Description
                      </th>
                    </IndexListHeaderCell>
                    <IndexListHeaderCell
                      alignment="left"
                      columnName="Status"
                    >
                      <th
                        className="index-list--header-cell index-list--align-left"
                        onClick={[Function]}
                        style={
                          Object {
                            "width": undefined,
                          }
                        }
                      >
                        Status
                      </th>
                    </IndexListHeaderCell>
                  </tr>
                </thead>
              </IndexListHeader>
              <IndexListBody
                columnCount={2}
                emptyState={
                  <EmptyState
                    size="lg"
                  >
                    <EmptyStateText
                      text="There are not any Tokens associated with this account. Contact your administrator"
                    />
                  </EmptyState>
                }
              >
                <tbody
                  className="index-list--body"
                >
                  <TokenRow
                    auth={
                      Object {
                        "description": "im a token",
                        "id": "030444b11fb10000",
                        "links": Object {
                          "self": "/api/v2/authorizations/030444b11fb10000",
                          "user": "/api/v2/users/030444b10a710000",
                        },
                        "orgID": "030444b10a713000",
                        "permissions": Array [
                          Object {
                            "action": "write",
                            "resource": Object {
                              "type": "orgs",
                            },
                          },
                          Object {
                            "action": "write",
                            "resource": Object {
                              "type": "buckets",
                            },
                          },
                        ],
                        "status": "active",
                        "token": "ohEmfY80A9UsW_cicNXgOMIPIsUvU6K9YcpTfCPQE3NV8Y6nTsCwVghczATBPyQh96CoZkOW5DIKldya6Y84KA==",
                        "user": "watts",
                        "userID": "030444b10a710000",
                      }
                    }
                    key="030444b11fb10000"
                    onClickDescription={[Function]}
                  >
                    <IndexListRow
                      disabled={false}
                    >
                      <tr
                        className="index-list--row"
                      >
                        <IndexListRowCell
                          alignment="left"
                          revealOnHover={false}
                        >
                          <td
                            className="index-list--row-cell index-list--align-left"
                          >
                            <div
                              className="index-list--cell"
                            >
                              <a
                                data-test="token-description-030444b11fb10000"
                                href="#"
                                onClick={[Function]}
                              >
                                im a token
                              </a>
                            </div>
                          </td>
                        </IndexListRowCell>
                        <IndexListRowCell
                          alignment="left"
                          revealOnHover={false}
                        >
                          <td
                            className="index-list--row-cell index-list--align-left"
                          >
                            <div
                              className="index-list--cell"
                            >
                              active
                            </div>
                          </td>
                        </IndexListRowCell>
                        <IndexListRowCell
                          alignment="right"
                          revealOnHover={true}
                        >
                          <td
                            className="index-list--row-cell index-list--show-hover index-list--align-right"
                          >
                            <div
                              className="index-list--cell"
                            >
                              <ComponentSpacer
                                align="right"
                              >
                                <div
                                  className="component-spacer component-spacer--right component-spacer--horizontal"
                                >
                                  <Button
                                    active={false}
                                    color="danger"
                                    shape="none"
                                    size="xs"
                                    status="default"
                                    text="Delete"
                                    type="button"
                                  >
                                    <button
                                      className="button button-xs button-danger"
                                      disabled={false}
                                      tabIndex={0}
                                      title="Delete"
                                      type="button"
                                    >
                                      Delete
                                    </button>
                                  </Button>
                                </div>
                              </ComponentSpacer>
                            </div>
                          </td>
                        </IndexListRowCell>
                      </tr>
                    </IndexListRow>
                  </TokenRow>
                  <TokenRow
                    auth={
                      Object {
                        "description": "im a token",
                        "id": "1",
                        "links": Object {
                          "self": "/api/v2/authorizations/030444b11fb10000",
                          "user": "/api/v2/users/030444b10a710000",
                        },
                        "orgID": "030444b10a713000",
                        "permissions": Array [
                          Object {
                            "action": "write",
                            "resource": Object {
                              "type": "orgs",
                            },
                          },
                          Object {
                            "action": "write",
                            "resource": Object {
                              "type": "buckets",
                            },
                          },
                        ],
                        "status": "active",
                        "token": "ohEmfY80A9UsW_cicNXgOMIPIsUvU6K9YcpTfCPQE3NV8Y6nTsCwVghczATBPyQh96CoZkOW5DIKldya6Y84KA==",
                        "user": "watts",
                        "userID": "030444b10a710000",
                      }
                    }
                    key="1"
                    onClickDescription={[Function]}
                  >
                    <IndexListRow
                      disabled={false}
                    >
                      <tr
                        className="index-list--row"
                      >
                        <IndexListRowCell
                          alignment="left"
                          revealOnHover={false}
                        >
                          <td
                            className="index-list--row-cell index-list--align-left"
                          >
                            <div
                              className="index-list--cell"
                            >
                              <a
                                data-test="token-description-1"
                                href="#"
                                onClick={[Function]}
                              >
                                im a token
                              </a>
                            </div>
                          </td>
                        </IndexListRowCell>
                        <IndexListRowCell
                          alignment="left"
                          revealOnHover={false}
                        >
                          <td
                            className="index-list--row-cell index-list--align-left"
                          >
                            <div
                              className="index-list--cell"
                            >
                              active
                            </div>
                          </td>
                        </IndexListRowCell>
                        <IndexListRowCell
                          alignment="right"
                          revealOnHover={true}
                        >
                          <td
                            className="index-list--row-cell index-list--show-hover index-list--align-right"
                          >
                            <div
                              className="index-list--cell"
                            >
                              <ComponentSpacer
                                align="right"
                              >
                                <div
                                  className="component-spacer component-spacer--right component-spacer--horizontal"
                                >
                                  <Button
                                    active={false}
                                    color="danger"
                                    shape="none"
                                    size="xs"
                                    status="default"
                                    text="Delete"
                                    type="button"
                                  >
                                    <button
                                      className="button button-xs button-danger"
                                      disabled={false}
                                      tabIndex={0}
                                      title="Delete"
                                      type="button"
                                    >
                                      Delete
                                    </button>
                                  </Button>
                                </div>
                              </ComponentSpacer>
                            </div>
                          </td>
                        </IndexListRowCell>
                      </tr>
                    </IndexListRow>
                  </TokenRow>
                </tbody>
              </IndexListBody>
            </table>
          </IndexList>
          <OverlayTechnology
            visible={false}
          >
            <div
              className="overlay-tech"
            >
              <div
                className="overlay--dialog"
                data-test="overlay-children"
              />
              <div
                className="overlay--mask"
              />
            </div>
          </OverlayTechnology>
        </TokenList>
      </FilterList>
    </Spinner>
  </ResourceFetcher>
</Tokens>
`;
